src/test/java/pl/edu/agh/firecell/engine/algorithm/BasicAlgorithmTest.java

```diff
@@ -0,0 +1,82 @@
+package pl.edu.agh.firecell.engine.algorithm;
+
+import org.joml.Vector3i;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import pl.edu.agh.firecell.model.Cell;
+import pl.edu.agh.firecell.model.Material;
+import pl.edu.agh.firecell.model.State;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Stream;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+class BasicAlgorithmTest {
+
+    @ParameterizedTest(name = "Compute")
+    @MethodSource("engineComputeTestCases")
+    public void flattenIndexTest(State state, Vector3i cellIndex, Cell resultCell, BasicAlgorithm algorithm) {
```

Comment:
Method name should be `computeTest`, and method source should be `computeTestMethodSource`

Reply:
I don't understand convencion here, I mean in another test name="Index {0} should be flattened to {2} in {1} space."